documentation/modules/tracing/proc-instrumenting-producers-consumers.adoc

```diff
@@ -0,0 +1,288 @@
+  <version>{OpenTelemetryVersionLower}</version>
+</dependency>
+<dependency>
+  <groupId>io.opentelemetry</groupId>
+  <artifactId>opentelemetry-semconv</artifactId>
+  <version>{OpenTelemetryVersionLower}</version>
+</dependency>
+<dependency>
+  <groupId>io.opentelemetry</groupId>
+  <artifactId>opentelemetry-sdk-trace</artifactId>
+  <version>{OpenTelemetryVersionStableLower}</version>
+</dependency>
+<dependency>
+  <groupId>io.opentelemetry.instrumentation</groupId>
+  <artifactId>opentelemetry-kafka-clients-{OpenTelemetryKafkaClient}</artifactId>
+  <version>{OpenTelemetryVersionLower}</version>
+</dependency>
+<dependency>
+  <groupId>io.opentelemetry.instrumentation</groupId>
+  <artifactId>opentelemetry-kafka-clients-common</artifactId>
+  <version>{OpenTelemetryVersionLower}</version>
+</dependency>
+<dependency>
+  <groupId>io.opentelemetry</groupId>
+  <artifactId>opentelemetry-exporter-jaeger</artifactId>
+  <version>{OpenTelemetryVersionStableLower}</version>
+</dependency>
+<dependency>
+  <groupId>io.grpc</groupId>
+  <artifactId>grpc-netty-shaded</artifactId>
+  <version>{OpenTelemetryGrpcVersionLower}</version>
+</dependency>
+----
+--
+
+. Instrument your client application code using either a Decorator pattern or Interceptors.
+
+* The Decorator pattern instrumentation is different for OpenTracing and OpenTelemetry,
+but in both cases you add tracing to an instance of a producer or consumer.
++
+--
+.Example Decorator instrumentation for OpenTracing
+[source,java,subs=attributes+]
+----
+// Create an instance of the KafkaProducer:
+KafkaProducer<Integer, String> producer = new KafkaProducer<>(senderProps);
+
+// Create an instance of the TracingKafkaProducer:
+TracingKafkaProducer<Integer, String> tracingProducer = new TracingKafkaProducer<>(producer,
+        tracer);
+
+// Send:
+tracingProducer.send(...);
+
+// Create an instance of the KafkaConsumer:
+KafkaConsumer<Integer, String> consumer = new KafkaConsumer<>(consumerProps);
+
+// Create an instance of the TracingKafkaConsumer:
+TracingKafkaConsumer<Integer, String> tracingConsumer = new TracingKafkaConsumer<>(consumer,
+        tracer);
+
+// Subscribe:
+tracingConsumer.subscribe(Collections.singletonList("mytopic"));
+
+// Get messages:
+ConsumerRecords<Integer, String> records = tracingConsumer.poll(1000);
+
+// Retrieve SpanContext from polled record (consumer side):
+ConsumerRecord<Integer, String> record = ...
+SpanContext spanContext = TracingKafkaUtils.extractSpanContext(record.headers(), tracer);
+----
+
+.Example Decorator instrumentation for OpenTelemetry
+[source,java,subs=attributes+]
+----
+// Create an instance of the KafkaProducer with tracing:
+Producer<String, String> op = new KafkaProducer<>(
+            configs,
+            new StringSerializer(),
+            new StringSerializer()
+        );
+       Producer<String, String> producer = tracing.wrap(op)
+       KafkaTracing tracing = KafkaTracing.create(GlobalOpenTelemetry.get());
```

Comment:
This line needs to be before previous line -- you need tracing instance before you can use it.